README.md

```diff
@@ -168,3 +168,7 @@ Once a new version dir has been created and added to `supported.json`, it is tim
 6. Create a PR and get it approved. 
 7. Merge main (This will push the updates to S3).
 8. Browse to https://jslib.k6.io/{lib_name}/{desired_version}/index.js
+
+## Contributing
+
+Because the jslib repository essentially hosts versioned build files, we expect most of the review of the actual code producing said build files to be done upstream in each jslib's project dedicated repositories. Therefore, to streamline the process of publishing new jslib versions, and while we still encourage contributors to seek relevant reviews, pull requests do not require approvals to be merged. We expect contributors to act responsibly and not merge Pull Requests that might be harmful to the k6 ecosystem, nor merge Pull Requests from other contributors without explicit approval.
```

Comment:
:thinking: does that mean that any contributor will merge anything without the approval of any maintainer, and it will be deployed? :thinking:

Reply:
That's my understanding of what was decided by the k6 team: this repository mostly holds build files, we should also move whatever actual js code is in here to dedicated repositories, and we should give the flexibility to contributors to publish new versions of their jslib without having us blocking them.

Now, I also don't necessarily feel at ease with this change as is:  it could lead to somewhat of a security issue. Considering that anyone could push whatever package they wish without any other watch, they could also push malicious code (although I'm not certain what the scope and attack vectors could be in the context of k6+goja+jslib), but still.

Maybe a safer first approach is to indeed add "official" maintainers, and later add a bot such as the one in the definitely typed repo which checks that a new version's PR satisfies some predicate, pings maintainers for review, and allows the person to merge once.

Another alternative would be to allow only users from within the Grafana organization to push with reviews and to still require a review for externals.

What Do You Think?

Reply:
Yes, you got my concern right :hugs: 

>That's my understanding of what was decided by the k6 team:

For some reason I had the impression that we are extending the maintainers' list or allowing merging PRs without approvals only for grafana/k6 members.

So whatever the policy will be, I believe that something that has production deployment should have a gatekeeper.

Reply:
Thanks for the clarification on your assumptions folks 🙇🏻 I'm glad I double checked 🤝

Reply:
from what I know nobody external can *merge* PRs so this seems like an unlikely thing to happen. 

Arguably though the section should be renamed as I don't think we have "contributors" - all the APIs here are done by us. And contributing will entail creating a different repo with the code and then asking us to host the final js files here. This seems unlikely to happen (and has not happen so far :thinking: ). It arguably is not even needed as the users can just host them on github (or wherever). Contributing something like that will also require documentation and so and so forth at which point I would argue we will push back on this unless we decide to adopt it - at which point it is no longer outside contribution ;).

tl;dr - can we explain that there are no "contributions" and that merging PR bumping versions is not required to have reviewers. Again only maintainers can merge PRs either way.

Reply:
>from what I know nobody external can merge PRs so this seems like an unlikely thing to happen.

If that's the case I'm absolutely fine with that. What confuses me is that it's a public repo and we introducing contributing section.

Reply:
@mstoykov those are fair points, I'm glad we got to (all together) clarify this.

>  What confuses me is that it's a public repo, and we introducing contributing section. So yeah, maybe we just need to stick with another name.

@olegbespalov I'd be interested in the reason leading to your confusion? I've actually double-checked, and most of the **public** [100 top repositories](https://github.com/EvanLi/Github-Ranking/blob/master/Top100/Top-100-stars.md) on GitHub do have contribution guidelines in their README (vue, react, TensorFlow, bootstrap, you name it).

If that's satisfying for everyone, moving forward with this PR I will:
* Remove the **contributing** section.
* Add a note somewhere in the description part of the README mentioning that PRs bumping versions of already supported jslibs do not need reviewers.
* Add a note somewhere in the description part of the README mentioning that PRs changing the jslib repository/webpages/project or adding a library would still need to be reviewed by a maintainer of the repo.

Now that doesn't really address the need I had which triggered this work initially: when submitting a PR that's not a version bump, who should be included as maintainers? As of today, this is unclear to me, and most PRs have been relying on @sniku or @ppcano, but on the other hand, with them being so busy, that (rightfully) doesn't really scale. Thus I would like to somehow clarify who should be the "maintainers" of the repo. In the sense of the people to turn to when needing a review.

I believe there are a bunch of options to clarify this down the road:
- [codeowner file](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners)
- "just" list one or two people to include as a reviewer as a default
- having a script automating picking k6 devs and frontend devs at random as reviewers, like in k6 (the frontend team on top of that).

Reply:
>I'd be interested in the reason leading to your confusion? I've actually double-checked, and most of the public [100 top repositories](https://github.com/EvanLi/Github-Ranking/blob/master/Top100/Top-100-stars.md) on GitHub do have contribution guidelines in their README (vue, react, TensorFlow, bootstrap, you name it).

All these projects that you mention have an established process of contribution to it. Just to be clear, by contributor here, I mean anybody outside Grafana. And I believe none of these projects allows us to merge anything without the explicit approval of the maintainer (like we advised in that PR). 

>If that's satisfying for everyone, moving forward with this PR I will:

all that you mentioned sounds good to me :+1: 

>Now that doesn't really address the need I had which triggered this work initially: when submitting a PR that's not a version bump, who should be included as maintainers? 

I had an impression during our core sync that you also volunteered for this, but maybe I'm wrong here. And maybe we can ask the frontend team if this will be something that they are also interested in reviewing as part of open-source initiatives.